### PR TITLE
clang-tidy modernize-use-emplace

### PR DESCRIPTION
**Pull request by est77**
Avoid a copy when adding objects to STL containers.

### CODE REVIEW

File: src/appleseed/foundation/meta/tests/test_fresnel.cpp [hunk 1]
```diff
@@ -271,15 +271,13 @@ TEST_SUITE(Foundation_Math_Fresnel)
         {
             const double eta = fit<size_t, double>(i, 0, PointCount - 1, 0.5, 2.0);
 
-            integral_points.push_back(
-                Vector2d(
+            integral_points.emplace_back(
                     eta,
-                    integrate_diffuse_fresnel_reflectance(eta)));
+                    integrate_diffuse_fresnel_reflectance(eta));
```
Comment (dictoon) [inline src/appleseed/foundation/meta/tests/test_fresnel.cpp]: Fix indentation.

File: src/appleseed/foundation/meta/tests/test_fresnel.cpp [hunk 2]
```diff
@@ -271,15 +271,13 @@ TEST_SUITE(Foundation_Math_Fresnel)
...
 
-            approx_points.push_back(
-                Vector2d(
+            approx_points.emplace_back(
                     eta,
-                    fresnel_internal_diffuse_reflectance(eta)));
+                    fresnel_internal_diffuse_reflectance(eta));
```
Comment (dictoon) [inline src/appleseed/foundation/meta/tests/test_fresnel.cpp]: Fix indentation.

File: src/appleseed/foundation/utility/benchmark/benchmarksuite.cpp [hunk 3]
```diff
@@ -268,10 +268,9 @@ void BenchmarkSuite::run(
                         stopwatch,
                         BenchmarkSuite::Impl::measure_runtime_ticks,
                         max<size_t>(1, measurement_count / 100));
-                points.push_back(
-                    Vector2d(
+                points.emplace_back(
                         static_cast<double>(j),
-                        ticks > overhead_ticks ? ticks - overhead_ticks : 0.0));
+                        ticks > overhead_ticks ? ticks - overhead_ticks : 0.0);
```
Comment (dictoon) [inline src/appleseed/foundation/utility/benchmark/benchmarksuite.cpp]: Fix indentation.

File: src/appleseed/renderer/kernel/intersection/assemblytree.cpp [hunk 4]
```diff
@@ -137,11 +137,10 @@ void AssemblyTree::collect_assembly_instances(
             continue;
 
         // Create and store an item for this assembly instance.
-        m_items.push_back(
-            Item(
+        m_items.emplace_back(
                 &assembly,
                 &assembly_instance,
-                cumulated_transform_seq));
+                cumulated_transform_seq);
```
Comment (dictoon) [inline src/appleseed/renderer/kernel/intersection/assemblytree.cpp]: Fix indentation.

File: src/appleseed/renderer/meta/tests/test_samplegeneratorjob.cpp [hunk 5]
```diff
@@ -55,10 +55,9 @@ TEST_SUITE(Renderer_Kernel_Rendering_Progressive_SampleGeneratorJob)
         for (uint64 x = 0; x < N; x += 100)
         {
             const uint64 y = SampleGeneratorJob::samples_to_samples_per_job(x);
-            points.push_back(
-                Vector2d(
+            points.emplace_back(
                     static_cast<double>(x),
-                    static_cast<double>(y)));
+                    static_cast<double>(y));
```
Comment (dictoon) [inline src/appleseed/renderer/meta/tests/test_samplegeneratorjob.cpp]: Fix indentation.

File: src/appleseed/renderer/meta/tests/test_sss.cpp [hunk 6]
```diff
@@ -389,15 +389,13 @@ TEST_SUITE(Renderer_Modeling_BSSRDF_SSS)
         {
             const float rd = fit<size_t, float>(i, 0, PointCount - 1, 0.0f, 1.0f);
 
-            std_points.push_back(
-                Vector2d(
+            std_points.emplace_back(
                     rd,
-                    compute_alpha_prime(std_rd_fun, rd)));
+                    compute_alpha_prime(std_rd_fun, rd));
```
Comment (dictoon) [inline src/appleseed/renderer/meta/tests/test_sss.cpp]: Fix indentation.

File: src/appleseed/renderer/modeling/object/meshobjectreader.cpp [hunk 7]
```diff
@@ -230,8 +230,7 @@ namespace
                 // Create the polygon to triangulate.
                 for (size_t i = 0; i < m_vertex_count; ++i)
                 {
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/object/meshobjectreader.cpp]: Can remove the curly braces now that the body fits on one line.

**Comment by dictoon** [comment]
Builds with VS 2012 so good to merge once review comments have been addressed!

**Comment by est77** [comment]
FIxed and rebased.

**Pull request merged**